iati/iati/custom_middleware.py

```diff
@@ -0,0 +1,20 @@
+"""Middleware for redirecting uppercase urls into lowercase."""
+from django import http
+from django.conf import settings
+
+
+class LowercaseMiddleware:
+    """Middleware class."""
+
+    def __init__(self, get_response):
+        """Initialise."""
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """Redirect url paths as lowercase except for documents or media files."""
+        response = self.get_response(request)
+        path = request.get_full_path()
+        lower = path.lower()
```

Comment:
I suggest a more descriptive variable name here - could become an issue that the variable name is the same as the string method `lower()`.

Reply:
Agreed, better keep variables and python's internals separate and have ours a bit more descriptive (`lowercase_path`)